### PR TITLE
audit handle for system user

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditServiceTest.kt [hunk 1]
```diff
@@ -34,12 +38,22 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AuditEvent a
...
+  private val service = AuditService(auditEventRepository, licenceRepository, communityOffenderManagerRepository)
 
   @BeforeEach
   fun reset() {
-    reset(auditEventRepository, licenceRepository)
+    val authentication = mock<Authentication>()
+    val securityContext = mock<SecurityContext>()
+    whenever(authentication.name).thenReturn("smills")
+    whenever(securityContext.authentication).thenReturn(authentication)
+    whenever(securityContext.authentication).thenReturn(authentication)
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditServiceTest.kt:51]: duplication here

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditServiceTest.kt [hunk 2]
```diff
@@ -662,6 +670,47 @@ class AuditServiceTest {
...
+        expandedConditionText = "updatedText",
+      )
+
+      whenever(communityOffenderManagerRepository.findByUsernameIgnoreCase("smills")).thenReturn(null)
+      service.recordAuditEventUpdateAdditionalConditionData(aLicenceEntity, anAdditionalCondition)
+
+      val auditCaptor = ArgumentCaptor.forClass(EntityAuditEvent::class.java)
+      verify(auditEventRepository, times(1)).save(auditCaptor.capture())
+
+      assertThat(auditCaptor.value.licenceId).isEqualTo(aLicenceEntity.id)
```
[the remark this answers is not in the document]
Comment (andrewrlee) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditServiceTest.kt:686]: e.g: https://stackoverflow.com/a/45873608

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt [hunk 3]
```diff
@@ -31,7 +30,6 @@ class LicenceConditionService(
   private val additionalConditionUploadDetailRepository: AdditionalConditionUploadDetailRepository,
   private val conditionFormatter: ConditionFormatter,
   private val licencePolicyService: LicencePolicyService,
-  private val communityOffenderManagerRepository: CommunityOffenderManagerRepository,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt:34]: this is nice - reduction in dependencies and removing behaviour unrelated to conditions out of here 👍

Comment (nishanthmahasamudram) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt:34]: Thank you.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditService.kt [hunk 4]
```diff
@@ -264,13 +265,23 @@ class AuditService(
...
+    return AuditEvent(
       licenceId = licence.id,
-      username = currentUser.username,
-      fullName = "${currentUser.firstName} ${currentUser.lastName}",
-      summary = "$summary for ${licence.forename} ${licence.surname}",
+      username = currentUser?.username ?: "SYSTEM",
+      fullName = if (currentUser != null) "${currentUser.firstName} ${currentUser.lastName}" else "SYSTEM",
+      eventType = if (currentUser == null) AuditEventType.SYSTEM_EVENT else AuditEventType.USER_EVENT,
+      summary =
+      if (currentUser != null) "$summary for ${licence.forename} ${licence.surname}" else "Unable to find offender for $authUsername",
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditService.kt:282]: don't think we need the warning message here - its more important to have the summary message so could leave this as is?

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**